src/gp_engine.hpp

```diff
@@ -143,13 +156,15 @@ class GPEngine {
             .array();
   }
 
-  inline void PreparePerPatternLogLikelihoods(size_t src1_idx, size_t src2_idx) {
-    per_pattern_log_likelihoods_ =
-        (plvs_.at(src1_idx).transpose() * transition_matrix_ * plvs_.at(src2_idx))
-            .diagonal()
-            .array()
-            .log() +
-        LogRescalingFor(src1_idx) + LogRescalingFor(src2_idx);
+  inline void PreparePerPatternLogLikelihoods(size_t gpcsp_idx, size_t src1_idx,
```

Comment:
Could we have a corresponding comment here about what these per-pattern log likes are, now that they aren't quite standard?

Reply:
I wrote some comments to indicate that these are actually computing marginal log likelihood over the trees that include the GPCSP corresponding to `gpcsp_idx`. 

Should we change the function name to reflect that? `PreparePerPatternMarginalLogLikelihoods` is probably going to cause confusion since these aren't marginal over all trees but rather a subset of the trees. `PreparePerPatternMarginalLogLikelihoodsOverGPCSPTrees`? Suggestions?

Reply:
I would suggest `PreparePerPatternLogLikelihoodsForGPCSP`. Thanks!

Reply:
Done.